portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -42,23 +42,16 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     String filter = request.getParameter("filter");
-    System.out.println(filter);
 
-    Query query;
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
-    if (filter.isEmpty()) {
-        System.out.println("filter is empty");
-        query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
-    } else {
-        System.out.println("filter is NOT empty");
+    if (!filter.equals("undefined") && !filter.isEmpty()) {
```

Comment:
I'd prefer to not send "undefined" as an expected value for the filter; rather, check in the JS whether `filter === undefined`, and send an empty string, if so. It's better to not use special values of the same type as normal values; maybe somebody has an online handle of "undefined" and wants to use that to leave a comment.

Reply:
That makes sense, I'll change this!